### PR TITLE
[0.12.2] Quality of Life Fixes

### PR DESCRIPTION
**Pull request by pixelpxed**
| Before | After |
| --- | --- |
| localhost_3000_en-US_teach(iPad Mini) (1) | localhost_3000_en-US_teach(iPad Mini) |

## Features
- Homeroom content preview in mobile view
  Related to [FRO-228](https://linear.app/skiso/issue/FRO-228/students-can-edit-homeroom-content)
- Removed schedule editor funny messages (Non-developers admins may not understand the joke, sorry)
- Disabled teacher's schedule editor `<SegmentedButton>` since it's not implemented (yet)
- Activity leave translation

## Fixes
- Teach page electives card session count ignores Academic Year, Semester
- Teach page subject list ignores Academic Year, Semester
- Admin schedule editor has no x-margin
- Attendance date picker not accounting for a special date
- Attendance homeroom content is editable to student admins
  Fixes [FRO-218](https://linear.app/skiso/issue/FRO-218/attendance-date-picker-not-accounting-for-special-dates)
- `{{time, time}}` in Classrooms page
  Fixes [FRO-214](https://linear.app/skiso/issue/FRO-214/time-time-in-classrooms-page)
- Students can edit homeroom content
  Fixes [FRO-228](https://linear.app/skiso/issue/FRO-228/students-can-edit-homeroom-content)

## Code Changes
- Updated environment template
- Updated package dependencies
  - Resolves `6 high severity vulnerabilities`
- Prettier touched files
- Added `911` as a valid classRegex

### CODE REVIEW

File: components/attendance/TodaySummary.tsx [hunk 1]
```diff
@@ -122,16 +124,29 @@ const TodaySummary: StylableFC<{
...
-                  <Button
-                    appearance="filled"
-                    icon={<MaterialIcon icon="add" />}
-                    onClick={() => setHomeroomView(HomeroomView.edit)}
-                    className="!justify-self-start"
-                  >
-                    {t("homeroom.action.add")}
-                  </Button>
+                  <>
+                    {mysk.user?.role === "student" && !mysk.user?.is_admin ? (
```
Comment (Sun-52) [inline components/attendance/TodaySummary.tsx:131]: The condition of whether to show the add/edit button in this component is different from the condition in the `editable` variable in `pages/classes/[classNumber]/attendance/date/[date].tsx`. This results in admin student being able to add/edit homeroom content in larger screen, while only being able to see in smaller screen. 
image
image

**Review by Sun-52 — COMMENTED** [review]
Not a very serious problem I think, but just a small change is needed.

File: components/attendance/TodaySummary.tsx (hunk 1 above)
Comment (pixelpxed) [reply components/attendance/TodaySummary.tsx:131]: I 'think' I forgot to use the editable variable, will fix that.

Comment (pixelpxed) [reply components/attendance/TodaySummary.tsx:131]: Good catch! Will fix that one.

**Review by Sun-52 — APPROVED** [review]
When adding / editing a homeroom content in small screens, you will have to refresh the page to be able to edit it again (the user can't type in the text box). I still don't know why this happens. 

It is probably not a very serious problem, so LGTM!

File: components/attendance/TodaySummary.tsx [hunk 2]
```diff
@@ -36,9 +37,19 @@ const TodaySummary: StylableFC<{
...
+  attendances,
+  homeroomContent,
+  classroomID,
+  editable,
+  style,
+  className,
+}) => {
   const { t } = useTranslation("attendance", { keyPrefix: "day" });
 
+  const mysk = useMySKClient();
```
[the remark this answers is not in the document]
Comment (pixelpxed) [reply components/attendance/TodaySummary.tsx:51]: It was used to check if the user can edit homeroom content or not, it was first implemented by a condition checked by MySK client itself. But we changed it to pass down the editable argument from the parent element instead.

TL;DR : I'll remove that, it was not used anymore.

**Pull request merged**